### ISSUE TITLE
Setting `proxyListenerMode` to `multiplex` generates an invalid config in the teleport-cluster Helm chart

### ISSUE THREAD

**Issue by pzduniak** [gravitational/teleport#9753]
### Description

**What happened**:

I'm trying literally anything to get app access running. I have a serious issue with certificates (for some reason they're ignoring the wildcard?), so I'm testing other configs. TLS multiplexing sounds related, so I enabled it. It completely broke the cluster.

**What you expected to happen**:

That it should start up.

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. Install a helm chart with the following values:
```
      clusterName: "teleport.realdomain.com",
      annotations: {
        service: {
          'external-dns.alpha.kubernetes.io/hostname': 'teleport.realdomain.com.,*.teleport.realdomain.com.'
        },
      },
      highAvailability: {
        certManager: {
          enabled: true,
          issuerName: 'dns01',
          issuerKind: 'ClusterIssuer',
        },
      },
      proxyListenerMode: 'multiplex'
```
2. ???
3. It's broken.

### Server Details
- Teleport version (run `teleport version`): 8.0.7, both Teleport and the Helm chart
- Server OS (e.g. from `/etc/os-release`): Docker
- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): Oracle Kubernetes Engine
- Additional details:

### Client Details
Irrelevant

### Debug Logs
Not debug, but should be enough.
```
2022-01-12T13:30:06Z WARN             both kubernetes_service and proxy_service are enabled, but proxy_service doesn't set kube_listen_addr; consider setting kube_listen_addr on proxy_service, to handle incoming Kubernetes requests config/configuration.go:1005
2022-01-12T13:30:06Z INFO [PROC:1]    Service diag is creating new listener on 0.0.0.0:3000. service/signals.go:213
2022-01-12T13:30:06Z INFO [DIAG:1]    Starting diagnostic service on 0.0.0.0:3000. service/service.go:2353
2022-01-12T13:30:06Z INFO [AUTH]      Updating cluster networking configuration: Kind:"cluster_networking_config" Version:"v2" Metadata:<Name:"cluster-networking-config" Namespace:"default" Labels:<key:"teleport.dev/origin" value:"config-file" > > Spec:<KeepAliveInterval:300000000000 KeepAliveCountMax:3 ProxyListenerMode:Multiplex > . auth/init.go:432
2022-01-12T13:30:06Z INFO [AUTH]      Updating session recording configuration: Kind:"session_recording_config" Version:"v2" Metadata:<Name:"session-recording-config" Namespace:"default" Labels:<key:"teleport.dev/origin" value:"config-file" > > Spec:<Mode:"node" ProxyChecksHostKeys:"\010\001" > . auth/init.go:450
2022-01-12T13:30:06Z INFO [AUTH]      Updating cluster auth preference: AuthPreference(Type="local",SecondFactor="otp"). auth/init.go:414
2022-01-12T13:30:06Z INFO [AUTH]      Updating cluster configuration: StaticTokens([]). auth/init.go:307
2022-01-12T13:30:06Z INFO [AUTH]      Created namespace: "default". auth/init.go:314
2022-01-12T13:30:06Z INFO [AUTH]      Auth server is running periodic operations. auth/init.go:392
2022-01-12T13:30:07Z INFO [AUTH:1:CA] Cache "auth" first init succeeded. cache/cache.go:676
2022-01-12T13:30:07Z INFO [PROC:1]    Service auth is creating new listener on 0.0.0.0:3025. service/signals.go:213
2022-01-12T13:30:07Z INFO [AUTH:1]    Starting Auth service with PROXY protocol support. service/service.go:1335
2022-01-12T13:30:07Z WARN [AUTH:1]    Configuration setting auth_service/advertise_ip is not set. guessing 10.244.0.21:3025. service/service.go:1413
[AUTH]         Auth service 8.0.7:v8.0.7-0-geb8076446 is starting on 10.244.0.21:3025.
2022-01-12T13:30:07Z INFO [AUTH:1]    Auth service 8.0.7:v8.0.7-0-geb8076446 is starting on 10.244.0.21:3025. utils/cli.go:282
2022-01-12T13:30:07Z INFO [PROC:1]    Connecting to the cluster teleport.realdomain.com with TLS client certificate. service/connect.go:132
2022-01-12T13:30:07Z INFO [PROC:1]    Connecting to the cluster teleport.realdomain.com with TLS client certificate. service/connect.go:132
2022-01-12T13:30:07Z INFO [PROC:1]    Kube: features loaded from auth server: Kubernetes:true App:true DB:true  service/connect.go:59
2022-01-12T13:30:07Z INFO [PROC:1]    Proxy: features loaded from auth server: Kubernetes:true App:true DB:true  service/connect.go:59
2022-01-12T13:30:07Z INFO [KUBERNETE] Cache "kube" first init succeeded. cache/cache.go:676
2022-01-12T13:30:07Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log. service/service.go:2105
2022-01-12T13:30:07Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log/upload. service/service.go:2105
2022-01-12T13:30:07Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log/upload/sessions. service/service.go:2105
2022-01-12T13:30:07Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log/upload/sessions/default. service/service.go:2105
2022-01-12T13:30:07Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log. service/service.go:2105
2022-01-12T13:30:07Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log/upload. service/service.go:2105
2022-01-12T13:30:07Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log/upload/streaming. service/service.go:2105
2022-01-12T13:30:07Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log/upload/streaming/default. service/service.go:2105
2022-01-12T13:30:07Z WARN [PROC:1]    Teleport process has exited with error. error:[
ERROR REPORT:
Original Error: *trace.BadParameterError set kubernetes_service.listen_addr if this process can be reached from a teleport proxy or point teleport.auth_servers to a proxy to dial out
Stack Trace:
        /go/src/github.com/gravitational/teleport/lib/service/kubernetes.go:115 github.com/gravitational/teleport/lib/service.(*TeleportProcess).initKubernetesService
        /go/src/github.com/gravitational/teleport/lib/service/kubernetes.go:62 github.com/gravitational/teleport/lib/service.(*TeleportProcess).initKubernetes.func1
        /go/src/github.com/gravitational/teleport/lib/service/supervisor.go:494 github.com/gravitational/teleport/lib/service.(*LocalService).Serve
        /go/src/github.com/gravitational/teleport/lib/service/supervisor.go:263 github.com/gravitational/teleport/lib/service.(*LocalSupervisor).serve.func1
        /opt/go/src/runtime/asm_amd64.s:1581 runtime.goexit
User Message: set kubernetes_service.listen_addr if this process can be reached from a teleport proxy or point teleport.auth_servers to a proxy to dial out] service:kube.init service/supervisor.go:268
2022-01-12T13:30:07Z WARN [KUBERNETE] Re-init the cache on error: watcher is closed: rpc error: code = Canceled desc = grpc: the client connection is closing. cache/cache.go:725
2022-01-12T13:30:07Z ERRO [PROC:1]    Critical service kube.init has exited with error set kubernetes_service.listen_addr if this process can be reached from a teleport proxy or point teleport.auth_servers to a proxy to dial out, aborting. service/signals.go:144
ERROR: set kubernetes_service.listen_addr if this process can be reached from a teleport proxy or point teleport.auth_servers to a proxy to dial out
```
None of this seems to be set by `proxyListenerMode ` in https://github.com/gravitational/teleport/blob/master/examples/chart/teleport-cluster/templates/config.yaml

**Comment by webvictim** [comment]
This is addressed by #10002

**Comment by ClementGautier** [comment]
For people coming here for solutions, there is 2 workarounds at the moment: 
1. Use `chartMode: custom`, copy the confimap before-hand and modify the teleport.yaml configuration to add `listen_addr: 0.0.0.0:443` on the `kubernetes_service` section.
2. or Fork the chart, do the fix proposed in the linked PR and use this custom chart
